client_cli/pytests/README.md

```diff
@@ -1,92 +1,164 @@
+
+---
+
+- **Accounts**: Test cases for account-related operations, such as account registration, key management, and metadata manipulation.
+- **Assets**: Test cases for asset-related operations, including asset creation, minting, burning, transferring, and managing asset definitions and metadata.
+- **Atomicity**: Test cases for transaction atomicity, including multiple instructions within a single transaction, paired instructions, and invalid instructions.
+- **Domains**: Test cases for domain-related operations, such as registering and unregistering domains.
+- **Roles**: Test cases for roles management.
+
+---
+ALT:
+- **Accounts**: Test cases for account-related operations:
+	- `test_accounts_query_filters.py` — various accounts-related filter queries.
+	- `test_register_accounts.py` — account registration.
+	- `test_set_key_value_pair.py` — key management.
+- **Assets**: Test cases for asset-related operations:
+	- `test_assets_query_filters.py` — various assets-related filter queries.
+	- `test_burn_assets.py` — burning assets.
+	- `test_mint_assets.py` — minting assets.
+	- `test_register_asset_definitions.py` — registering various types of asset definitions.
+	- `test_transfer_assets.py` — transferring assets.
+- **Atomicity**: Test cases for transaction atomicity:
+	- `test_multiple_instructions_within_transaction.py` — multiple instructions within a single transaction.
+	- `test_pair_instructions_within_transaction.py` — paired instructions within a single transaction.
+	- `test_wrong_instructions_within_transaction.py` — invalid instructions within a single transaction.
+- **Domains**: Test cases for domain-related operations:
+	- `test_domains_query_filters.py` — various domain-related filter queries.
+	- `test_register_domains.py` — registering various types of domains.
+	- `test_transfer_domains.py` — transferring a domain.
+- **Roles** Test cases for roles management:
+	- `test_register_roles.py` — registering a role, attaching permissions to a role, granting a role to an account.
+
+## Using Test Suites
+
+1. Set up a test environment using the [`test_env.py`](../../scripts/test_env.py) file:
+
+	 ```shell
+	 # Must be executed from the repo root:
+	 ./scripts/test_env.py setup
+	 ```
+
+   By default, this builds `iroha`, `iroha_client_cli`, and `kagami`, and runs four peers with their API exposed through the `8080`-`8083` ports.\
+	 This behavior can be reconfigured. You can run `./scripts/test_env.py --help` to see the list of available commands and options.
+
+2. Install and configure [Poetry](https://python-poetry.org/). For details, see [Poetry Configuration](#poetry-configuration) below.
+3. Configure the tests. For details, see [Tests Configuration](#tests-configuration) below.
+4. Run the tests:
+
+	 ```shell
+	 poetry run pytest
+	 ```
+
+5. Clean up the test environment:
+
+	 ```shell
+	 # Must be executed from the repo root:
+	 ./scripts/test_env.py cleanup
+	 ```
+
```

Comment:
Here we can put a _Custom Environment_ section, as discussed in https://github.com/hyperledger/iroha/pull/4157#discussion_r1430840668